apps/cowswap-frontend/src/modules/account/containers/AccountDetails/index.tsx

```diff
@@ -228,26 +234,36 @@ export function AccountDetails({
             <WalletSecondaryActions>
               {!isInjectedMobileBrowser && (
                 <>
-                  <WalletAction onClick={handleDisconnectClick}>
-                    <Trans>Disconnect</Trans>
-                  </WalletAction>
+                  {account && !isChainIdUnsupported && (
+                    <AddressLink
+                      hasENS={!!ENSName}
+                      isENS={!!ENSName}
+                      href={getEtherscanLink(chainId, 'address', ENSName ? ENSName : account)}
+                    >
+                      {explorerLabel} ↗
+                    </AddressLink>
+                  )}
+
+                  {/* // TODO: Check if isHardWareWallet + is Trezor */}
+                  {connection.type !== ConnectionType.GNOSIS_SAFE && isHardWareWallet && (
```

Comment:
How can it be Hardware wallet + Gnosis Safe? 

Also, we can easily tell if is just trezor by checking the connectionType.type against `ConnectionType.TREZOR`

Reply:
Fixed the condition